### PR TITLE
Bump and pin swc again, pin more swc crates

### PR DESCRIPTION
**Pull request by pyrrho**
This is an alternative to DioxusLabs/dioxus#2861 and should close DioxusLabs/dioxus#2878

### CODE REVIEW

**Comment by pyrrho** [comment]
I believe this will fix the install issue, but I'm not sure how to confirm that, because I can't seem to reproduce the failure when running `cargo build -p dioxus-cli --release` inside of a local clone of Dioxus checked out at the `v0.6.0-alpha.2` tag. :-/

**Comment by pyrrho** [comment]
> `cargo install` ignores the lock file by default

Hence installing with `--locked` being the fix, right.
Man. What a choice that was...

-----

> You need to run `cargo update` in the local repo

That got me reproducing the install error, and let me set up a [cli_install_fix_repro](https://github.com/pyrrho/dioxus/tree/cli_install_fix_repro) branch to test locally.

There are a lot of legitimate build failures, more than are showing up in CI... I'll take a look at what it'll take to update manganis to use the new swc API, but it might take me a while to get done.

File: test-package/test-package-nested-dependency/src/lib.rs [hunk 1]
```diff
@@ -1,7 +1,7 @@
 mod file;
 pub use file::*;
 mod font;
-pub use font::*;
+// pub use font::*;
```
Comment (pyrrho) [inline test-package/test-package-nested-dependency/src/lib.rs:4]: Commenting this out doesn't feel right, but Clippy was mad about `pub use`ing a [file that contains nothing but comments](https://github.com/DioxusLabs/manganis/blob/main/test-package/test-package-nested-dependency/src/font.rs), so....

File: cli-support/src/file.rs [hunk 2]
```diff
@@ -210,6 +210,7 @@ pub(crate) fn minify_js(source: &AssetSource) -> anyhow::Result<String> {
                         mangle: BoolOrDataConfig::from_bool(true),
                         ..Default::default()
                     },
+                    JsMinifyExtras::default(),
```
Comment (pyrrho) [inline cli-support/src/file.rs:213]: Some context on this Extras object:
https://github.com/swc-project/swc/issues/5386

File: test-package/test-package-nested-dependency/src/lib.rs (hunk 1 above)
Comment (ealmloff) [reply test-package/test-package-nested-dependency/src/lib.rs]: Remote assets were removed and then restored, but I guess the tests were never restored

**Review by ealmloff — APPROVED** [review]

**Pull request merged**